rtslib/root.py

```diff
@@ -278,26 +278,32 @@ def clear_existing(self, confirm=False):
 
         # Targets depend on storage objects, delete them first.
         for t in self.targets:
-            t.delete()
+            if (not storage_object and not target) or (target and t.wwn == target):
+                t.delete()
+
         for fm in (f for f in self.fabric_modules if f.has_feature("discovery_auth")):
             fm.clear_discovery_auth_settings()
+
         for so in self.storage_objects:
-            so.delete()
+            if (not storage_object and not target) or (storage_object and so.name == storage_object):
+                so.delete()
 
         # If somehow some hbas still exist (no storage object within?) clean
         # them up too.
-        for hba_dir in glob.glob("%s/core/*_*" % self.configfs_dir):
-            os.rmdir(hba_dir)
+        if not (storage_object or target):
+            for hba_dir in glob.glob("%s/core/*_*" % self.configfs_dir):
+                os.rmdir(hba_dir)
```

Comment:
Here should we also remove the configfs_dir related to the "storage_object" and "target" ?

Reply:
@lxbsz this is not needed now, especially for single delete's case, as this is trying to delete stale hba dirs if any.
 
https://github.com/open-iscsi/rtslib-fb/pull/118 will anyway delete them if any stales exists before create starts.

Reply:
Okay, that's cool.
Thanks.

Reply:
@lxbsz yeah, I thought it was a rare case, so left it there :-)

But yeah, since you have pointed it, it won't hurt addressing it. Will do.

And in python there is no goto :) which would have mad life easy.